Added FreeRTOS and ESP32 Core settings to worker configuration

rmontrosecbw opened the pull request:
Added the ability to set FreeRTOS task priority setting for workers, and for ESP32 devices, assigning the worker task to a specific CPU core.
```js
let aWorker = new Worker("simpleworker", {
	static: 8192,		// bytes
	stack: 64,			// JavaScript stack in 16 byte slots
	heap: {				// Values in 16 byte slots
		initial: 64,
		incremental: 32
	},
	nativeStack: 8192,	// host's native "C" stack, in bytes
	priority: 3,		// For FreeRTOS, sets the Worker Task priority
	coreId: 1			// For ESP32, set the execution core. 0 or 1
});
```
Enhanced `worker.md` documentation to reflect the changes.

rmontrosecbw committed:
Worker subclass Example

Showing how to create a class derived from the Worker class. You need a separate coreworker.js function for the task.

rmontrosecbw committed:
Adds option to pin worker to CPU core in ESP32 multicore systems.

Documentation explains more about setting native stack size for workers on FreeRTOS based systems.

rmontrosecbw committed:
Added setting worker task priority. Updated modWorker.c and worker.md documentation

rmontrosecbw committed:
Added priority to xsCreation structure

phoddie commented:
The priority and core additions are welcome. They may eventually be useful on other platforms.

A couple of items to correct:

- No need to extend the native `xsCreation` structure (and definitely not across every platform XS runs on). It looks like the new properties could be local variables in `workerConstructor()`. 
- `coreId` should be `core` – the `Id` doesn't add information.

rmontrosecbw committed:
- Moved priority and core out of xsCreation; renamed coreId to core
- Added additional trace to worker example to better show the operation of the two tasks, have core and priority set in example (only effective on FreeRTOS and ESP32 systems)

mkellner commented:
This has been merged and will be available in the next release.
